src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -0,0 +1,345 @@
+      actor->SetNodeID(ClientID::Nil());
+      // Since the actor is not yet bind with leased worker, so we should enqueue the
+      // actor otherwise it will never be created later.
+      Enqueue(actor);
+    }
+  });
+}
+
+void GcsActorManager::LeaseWorker(std::shared_ptr<GcsActor> actor,
+                                  std::shared_ptr<GcsNode> node) {
+  RAY_CHECK(actor && node);
+  RAY_LOG(DEBUG) << "Start leasing worker from node " << node->GetNodeId()
+                 << " for actor " << actor->GetActorID();
+
+  std::weak_ptr<GcsNode> weak_node(node);
+  node->LeaseWorker(actor, [this, actor, weak_node](
+                               const Status &status, const ClientID &spillback_to,
+                               std::shared_ptr<GcsLeasedWorker> worker) {
+    if (auto node = weak_node.lock()) {
+      if (!status.ok()) {
+        if (auto gcs_node_manager = gcs_node_manager_.lock()) {
+          // TODO(zsl): Treat it as NodeFailure when !status.ok(), it may be a little
+          // strict. Maybe better to retry leasing worker from this node.
+          gcs_node_manager->OnNodeFailure(node->GetNodeId());
+        }
+        return;
+      }
+
+      if (!spillback_to.IsNil()) {
+        RAY_CHECK(worker == nullptr);
+        actor->SetNodeID(spillback_to);
+        AsyncFlush(actor, [this, actor] { Enqueue(actor); });
+        return;
+      }
+
+      RAY_LOG(INFO) << "Succeed in leasing worker " << worker->GetWorkerID() << " from "
+                    << worker->GetNodeID() << " for actor " << actor->GetActorID();
+      OnWorkerLeased(actor, std::move(worker));
+    }
+  });
+}
+
+void GcsActorManager::OnWorkerLeased(std::shared_ptr<GcsActor> actor,
+                                     std::shared_ptr<GcsLeasedWorker> worker) {
+  RAY_CHECK(actor && worker && actor->GetWorker() == nullptr);
+  RAY_LOG(INFO) << "Start creating actor " << actor->GetActorID() << " on worker "
+                << worker->GetWorkerID() << " at node " << actor->GetNodeID();
+  std::weak_ptr<GcsLeasedWorker> weak_worker(worker);
+  worker->CreateActor(actor, [this, actor, weak_worker](const Status &status) {
+    if (auto worker = weak_worker.lock()) {
+      if (!status.ok()) {
+        RAY_LOG(WARNING) << "Failed to create actor " << actor->GetActorID()
+                         << " on worker " << worker->GetWorkerID() << " with node id "
+                         << worker->GetNodeID() << ", address " << worker->GetIpAddress()
+                         << ":" << worker->GetPort() << " as session is broken.";
+        if (auto gcs_node_manage = gcs_node_manager_.lock()) {
+          if (auto node = gcs_node_manage->GetNode(worker->GetNodeID())) {
+            // TODO(zsl): Treat it as WorkerFailure when !status.ok(), it may be a little
+            // strict too. Maybe better to retry creating actor on this worker.
+            node->OnWorkerFailure(worker->GetWorkerID());
```

Comment:
Why is this failure handler on a particular `node`? It seems to eventually just call back into the GcsActorManager, so why can't we just call `FailureActor` immediately? These callbacks are confusing and we should try to centralize the logic more instead.

Reply:
Just as I comment above, It is a little strict to think the worker is dead via the `!status.ok()`, It should retry `CreateActor` here.

The failure of worker should be detected only in place of HandleReportWorkerFailure inside worker_info_handler_impl.cc.

I will fix it.